### ISSUE TITLE
It is not possible to send multiple files in a message in Http

### ISSUE THREAD

**Issue by Jiralite** [serenity-rs/serenity#2924]
It does not seem possible to send multiple files in a message. See the following:

```Rust
use serenity::all::{ChannelId, CreateAttachment, Http};

#[tokio::main]
async fn main() {
    let client = Http::new("token");
    let mut files: Vec<CreateAttachment> = vec![];

    files.push(CreateAttachment::url(&client, "https://cdn.discordapp.com/attachments/1089254526642573343/1266824196982575134/bob.png?ex=66a68d79&is=66a53bf9&hm=d450fd822c5cef577dbc86322537a8e7b67fcc622650d3620ed830fbb57e6d59&").await.unwrap());
    files.push(CreateAttachment::url(&client, "https://cdn.discordapp.com/attachments/1089254526642573343/1266824054355267734/impl.png?ex=66a68d57&is=66a53bd7&hm=28cae6c5f20cd60d82dbeddf3fe11d3d64024d98ce4fe7b636b583eea51267da&").await.unwrap());

    println!("{:?}", files.len()); // 2

    let _ = client
        .send_message(
            ChannelId::new(channel_id),
            files,
            &{},
        )
        .await;
}
```

Only the first attachment will be sent.

It looks like the issue arises around here:

https://github.com/serenity-rs/serenity/blob/c7540cd3ee97662a7494516ce58a26c83fcf9419/src/builder/create_attachment.rs#L33-L40

The other functions lead to this one and the `id` is always `0`. This discards other buffers.

**Comment by arqunis** [comment]
Indeed, we do not support multiple files through raw `Http`, but this is unintentional. This commit https://github.com/serenity-rs/serenity/commit/2a81062598cfc999bcc0f2e9ccdfbc700f09f2a0 had refactored the code to allow editing existing attachments, but by doing so, it broke calling `Http::send_message` with your own attachments. This is because you're avoiding this function:

https://github.com/serenity-rs/serenity/blob/c7540cd3ee97662a7494516ce58a26c83fcf9419/src/builder/create_attachment.rs#L253-L256

It is called by builders such as `CreateMessage` to actually assign a proper ID to each attachment before handing them over to `Http::send_message`. So the support definitely exists, it's just unfortunately impossible with just `Http` alone. Try this instead:
```rust
use serenity::builder::CreateMessage;

let _ = CreateMessage::new().files(files).execute(&client, (ChannelId::new(channel_id), None)).await;
// or:
let _ = ChannelId::new(channel_id).send_message(&client, CreateMessage::new().files(files)).await;
```

**Comment by Jiralite** [comment]
> Try this instead:
> 
> ```rust
> use serenity::builder::CreateMessage;
> 
> let _ = CreateMessage::new().files(files).execute(&client, (ChannelId::new(channel_id), None)).await;
> // or:
> let _ = ChannelId::new(channel_id).send_message(&client, CreateMessage::new().files(files)).await;
> ```

This worked, so thank you for now!

**Comment by mkrasnitski** [comment]
This might be solved by #2771? Discord's attachment semantics are pretty complicated, but see if using the `next` branch fixes the behavior when using raw Http.

**Comment by Jiralite** [comment]
It looks like using the next branch worked!

<details>
<summary>Code</summary>

```Rust
// serenity = { git = "https://github.com/serenity-rs/serenity.git", branch = "next" }
use serenity::all::{ChannelId, CreateAttachment, Http};

#[tokio::main]
async fn main() {
    let client = Http::new("token");
    let mut files: Vec<CreateAttachment> = vec![];

    files.push(CreateAttachment::url(&client, "https://cdn.discordapp.com/attachments/1089254526642573343/1266824196982575134/bob.png?ex=66a887b9&is=66a73639&hm=fe104782c9f59b6834aaba6632b0daf460541750cd2c6ae204c45f789d3c9ab3&", "bob.png").await.unwrap());
    files.push(CreateAttachment::url(&client, "https://cdn.discordapp.com/attachments/1089254526642573343/1266816725836496908/image.png?ex=66a880c3&is=66a72f43&hm=a50c85ef377926f8b955fb66f473daeea4547426ced17103176753b1dedcc66a&", "image.png").await.unwrap());

    println!("{:?}", files.len()); // 2

    let _ = client
        .send_message(ChannelId::new(channel_id), files, &{})
        .await;
}
```
</details>

**Comment by ivinjabraham** [comment]
Looks like this issue was resolved by #2771. Perhaps we can close this now?

**Comment by jamesbt365** [comment]
Will close, doesn't seem like that much of a big deal to try and fix this on current.
